src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -94,12 +93,36 @@ public Objection createObjection(String requestId, String companyNumber, String
                 .withCreatedOn(dateTimeSupplier.get())
                 .withCreatedBy(new CreatedBy(ericUserId, userEmailAddress))
                 .withHttpRequestId(requestId)
+                .withActionCode(actionCode)
                 .withStatus(objectionStatus)
                 .build();
 
         return objectionRepository.save(entity);
     }
 
+    private Long getActionCode(String companyNumber, String requestId) {
+        Long actionCode = oracleQueryClient.getCompanyActionCode(companyNumber);
+
+        logger.debugContext(requestId, "Company action code is " + actionCode);
+
+        return actionCode;
+    }
+
+    private ObjectionStatus getObjectionStatusForCreate(Long actionCode, String companyNumber, String logContext) {
+        ObjectionStatus objectionStatus = ObjectionStatus.OPEN;
+        try {
+            actionCodeValidator.validate(actionCode, logContext);
```

Comment:
Might be a good idea to have a null check here if the action code is to be Long.

Reply:
Maybe I shouldn't have used a non-primitive type in my earlier PR to get the action code, thereby making it clear (enforcing) that non-null values are just not allowed?

Reply:
> Might be a good idea to have a null check here if the action code is to be Long.

if it is null, it should be validated as normal. The AllowedValuesValidationRule will throw a ValidationException containing ObjectionStatus.INELIGIBLE_NO_DISSOLUTION_ACTION if it is null, as null is not one of the codes specified in OBJ-209 (these codes are in config ACTION_CODES_STRIKE_OFF_NOTICE). I can add an extra unit test for that rule to test a null value being passed into it.

Reply:
> Maybe I shouldn't have used a non-primitive type in my earlier PR to get the action code, thereby making it clear (enforcing) that non-null values are just not allowed?

I think it's ok, it covers a scenario where a company has no action code

Reply:
I've added extra unit tests to the validation rules and they work as expected when passing null into them.

Reply:
> > Maybe I shouldn't have used a non-primitive type in my earlier PR to get the action code, thereby making it clear (enforcing) that non-null values are just not allowed?
> 
> I think it's ok, it covers a scenario where a company has no action code

Oh yes, true. And just seen that it is a nullable column in the database (although I did think that a company always had an action code).